walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,99 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.Collections;
+import java.util.Comparator;
 
 public final class FindMeetingQuery {
+
+  static final int startOfDay = TimeRange.START_OF_DAY;
+  static final int endOfDay = TimeRange.END_OF_DAY;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> allAttendees = new ArrayList<>();
+    allAttendees.addAll(request.getAttendees());
+    allAttendees.addAll(request.getOptionalAttendees());
+    MeetingRequest meetingWithEveryone = new MeetingRequest(allAttendees, request.getDuration());
+
+    Collection<TimeRange> meetingTimesWithAll = findTime(events, meetingWithEveryone);
+
+    // Returns meeting times based on if there are available times with optional attendees 
+    if (meetingTimesWithAll.size() > 0 || request.getAttendees().isEmpty()) {
+      return meetingTimesWithAll;
+    } else {
+      return findTime(events, request);
+    }
+  }
+
+  public Collection<TimeRange> findTime(Collection<Event> events, MeetingRequest request) {
+    
+    List<TimeRange> possibleMeetingTimes = new ArrayList<>();
+    List<TimeRange> blockedTimes = new ArrayList<>();
+    int requestDuration = (int)request.getDuration();
+    Collection<String> requestAttendees = request.getAttendees();
+    int numEvents = events.size();
+    int availableStart = startOfDay;
+    int counter = 0;
```

Comment:
It looks like counter is not used anymore

Reply:
removed